src/ModalBase/ModalPopup.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { FocusOn } from 'react-focus-on';
+import Portal from './Portal';
+import PopperElement from './PopperElement';
+import ModalContext from './ModalContext';
+
+const useMemoizedValues = ({ close, isOpen, isBlocking }) => useMemo(() => ({
```

Comment:
Heh, I asked you why you pulled this out earlier. This appears to be exactly the same, so maybe share it between the components?

Reply:
The ModalPopup and ModalLayer were in the same file as I was working. Forgot to work through what to do with this. I will DRY it up into a `ModalContextProvider`